daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXTypeCalcFunctions.scala

```diff
@@ -69,7 +71,7 @@ case class DFDLXOutputTypeCalcNextSibling(a: CompiledDPath, b: NodeInfo.Kind) ex
     val primType = nextSibling.erd.optPrimType.get
 
     val x = nextSibling.dataValue
-    typeCalculator.outputTypeCalcRun(dstate, x, primType)
+    typeCalculator.outputTypeCalcRun(dstate, x.getAnyRef, primType)
```

Comment:
As with the expressions, it seemed too complicated to make the typeCalculator aware of DataValue, so this is one of those boundaries where we cast to AnyRef